### PR TITLE
Waiting for running tiller pod in e2e

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/giantswarm/issues/7390

Tiller pod could be temporarily in outage after changing a tiller version. 
We cannot assume that we need to install a fresh tiller rather than upgrade by just checking once.

### CODE REVIEW

**Comment by tomahawk28** [comment]
Tested twice, no error came out.

File: ensure_tiller_installed.go [hunk 1]
```diff
@@ -197,11 +197,7 @@ func (c *Client) EnsureTillerInstalledWithValues(ctx context.Context, values []s
 	{
 		o := func() error {
 			pod, err = getPod(c.k8sClient, c.tillerNamespace)
-			if IsNotFound(err) {
```
Comment (tomahawk28) [inline ensure_tiller_installed.go]: Moving `IsNotFound` matcher usage outside of retries.

**Comment by tomahawk28** [comment]
I'm really not sure what makes failed at this test, https://circleci.com/gh/giantswarm/helmclient/1960?pipelines-ui-opt-out

**Comment by tfussell** [comment]
I see. All I can think of is adding a debug print statement to see what type the error is if it's not `IsNotFound` here https://github.com/giantswarm/helmclient/blob/f1d5a75b7dfc37edd4ca4f15de33a0368ea3d612/ensure_tiller_installed.go#L211

**Comment by tfussell** [comment]
The second build with extra logging seems different from the first even though the code didn't change. Can we run this locally to debug?

**Comment by tomahawk28** [comment]
It stranges, logs reports that tiller pod is even up-to-dated. Not sure what is broken.

**Comment by tomahawk28** [comment]
closing in favor of https://github.com/giantswarm/helmclient/pull/131

**Pull request closed without merging**